validator/core/src/lib.rs

```diff
@@ -0,0 +1,117 @@
+//! Core library for validating [Substrait](https://substrait.io/).
+//!
+//! Note that this crate is simply the greatest common denominator of the
+//! various language bindings for the validator, which doesn't include the
+//! remote YAML file resolution due to the complexities of linking against
+//! the system's cryptography/certificate validation libraries. Refer to
+//! the documentation for the `substrait_validator` crate for basic usage
+//! from within Rust.
+
+#[macro_use]
+pub mod output;
+
+#[macro_use]
+mod parse;
+
+pub mod export;
+pub mod input;
+
+// Aliases for common types used on the crate interface.
+pub type Config = input::config::Config;
```

Comment:
You can also pub use them:
```
mod input;
pub use input::config::Config;
...
```

Reply:
derp, that's much nicer. The C++ programmer in me didn't even consider trying to re-export something that isn't a "namespace."